### PR TITLE
feat: send notification when  pr opened

### PR DESCRIPTION
**Pull request by andreitraistaru**
## Reason

Add pr notification.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: .github/workflows/send_pr_notification.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Send notification on PR opened
+
+on:
+  pull_request_target:
+    types: [opened, reopened]
```
[the remark this answers is not in the document]
Comment (andreitraistaru) [reply .github/workflows/send_pr_notification.yml:5]: yes, sure, it should work if we use `types: [ labeled ]` and then introduce a check for the label like this `if: ${{ github.event.label.name == 'Status: Reviewer' }}`. I'll test it to confirm this.

File: .github/workflows/send_pr_notification.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+jobs:
+  send_notification:
+    runs-on: ubuntu-latest
+    steps:
+      - name: notify
+        run: |
+          curl -d '{ "GITHUB_PR_URL": "${{ github.event.pull_request.html_url }}" }' ${{ secrets.SLACK_PR_NOTIFICATION_WEBHOOK }}
```
Comment (pb8o) [inline .github/workflows/send_pr_notification.yml:13]: Could we also send deletions/additions? That way we could easily estimate how long it will take...

**Comment by JonathanWoollett-Light** [comment]
Since this doesn't affect Firecrackers functionality I don't think it should have the `feat` prefix, it may make more sense to use a commit title like [`chore: send notification when pr opened`](https://github.com/firecracker-microvm/firecracker/pull/3520/commits/59657c86e1c8959e2a39d3741e4341ae55880ac9) or [`ci: send notification when pr opened`](https://github.com/firecracker-microvm/firecracker/pull/3520/commits/59657c86e1c8959e2a39d3741e4341ae55880ac9).

File: .github/workflows/send_pr_notification.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (andreitraistaru) [reply .github/workflows/send_pr_notification.yml]: Should be implemented now.

File: .github/workflows/send_pr_notification.yml (hunk 2 above)
Comment (andreitraistaru) [reply .github/workflows/send_pr_notification.yml]: Should be implemented now.

**Review by mattschlebusch — APPROVED** [review]

**Comment by dianpopa** [comment]
It would be worth mentioning it in the commit title that we are doing this in order to get a slack notification.

**Review by roypat — APPROVED** [review]

File: .github/workflows/send_pr_notification.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  send_notification:
+    if: ${{ github.event.label.name }} == "Status:\ Reviewer"
+    runs-on: ubuntu-latest
+    steps:
+      - name: notify
+        run: |
+          curl -X POST ${{ secrets.SLACK_PR_NOTIFICATION_WEBHOOK }} \
+               -H 'Content-Type: application/json' \
+               -d '{ "GITHUB_PR_URL": "${{ github.event.pull_request.html_url }}",
+                     "PR_ADDITIONS": "${{ github.event.pull_request.additions }}",
```
Comment (dianpopa) [inline .github/workflows/send_pr_notification.yml:17]: what does this refer to? receive notif when new commits are added to a PR?

Comment (andreitraistaru) [reply .github/workflows/send_pr_notification.yml:17]: These are the fields related to the PR's metrics that Slack requires in order to show the message filled in with number of deletions, additions and changed files. For example, `PR_ADDITIONS` is the name of the field that Slack expects to find in the body of the POST in order to be able to send number of additions in the message we receive in Slack.

Comment (andreitraistaru) [reply .github/workflows/send_pr_notification.yml:17]: The name of these fields can be configured in the Slack workflow. Also the number of fields that should exist in the body of the POST is configurable. This is just my choice for getting this working according to current requirements.

**Review by pb8o — APPROVED** [review]

File: .github/workflows/send_pr_notification.yml (hunk 3 above)
Comment (dianpopa) [reply .github/workflows/send_pr_notification.yml:17]: ok, gotcha! now it is clear
thanks for the commit message update

**Review by roypat — APPROVED** [review]

**Pull request merged**